hera_cal/vis_clean.py

```diff
@@ -281,22 +307,419 @@ def write_data(self, data, filename, overwrite=False, flags=None, nsamples=None,
+            flags = self.flags
+        if keys is None:
+            keys = data.keys()
+        if wgts is None:
+            wgts = DataContainer(dict([(k, np.ones_like(flags[k], dtype=np.float)) for k in keys]))
+        # make sure flagged channels have zero weight, regardless of what user supplied.
+        wgts = DataContainer(dict([(k, (~flags[k]).astype(float) * wgts[k]) for k in keys]))
+        suppression_factors = [tol]
+        if max_frate is not None:
+            if isinstance(max_frate, (int, np.integer, float, np.float)):
+                max_frate = DataContainer(dict([(k, max_frate) for k in data]))
+            if not isinstance(max_frate, DataContainer):
+                raise ValueError("If fed, max_frate must be a float, or a DataContainer of floats")
+            # convert kwargs to proper units
+            max_frate = DataContainer(dict([(k, np.asarray(max_frate[k])) for k in max_frate]))
+
+        for k in keys:
+            if ax == 'freq' or ax == 'both':
+                filter_centers_freq = [0.]
+                bl_dly = self.bllens[k[:2]] * horizon + standoff / 1e9
+                filter_half_widths_freq = [np.max([bl_dly, min_dly / 1e9])]
+            if ax == 'time' or ax == 'both':
+                filter_centers_time = [0.]
+                if max_frate is not None:
+                    max_fr = max_frate[k] * 1e-3
+                    filter_centers_time = [0.]
+                    filter_half_widths_time = [max_fr]
+                else:
+                    raise ValueError("Must provide a maximum ringe-rate (or max frate dict) for time filtering.")
+            if ax == 'both':
+                filter_centers = [filter_centers_time, filter_centers_freq]
+                filter_half_widths = [filter_half_widths_time, filter_half_widths_freq]
+                filter_centers = [filter_centers_time, filter_centers_freq]
+                suppression_factors = [[tol], [tol]]
+            else:
+                suppression_factors = [tol]
+                if ax == 'freq':
+                    filter_centers = filter_centers_freq
+                    filter_half_widths = filter_half_widths_freq
+                elif ax == 'time':
+                    filter_centers = filter_centers_time
+                    filter_half_widths = filter_half_widths_time
+            self.fourier_filter(keys=[k], filter_centers=filter_centers, filter_half_widths=filter_half_widths,
+                                suppression_factors=suppression_factors, mode='dayenu_dpss_leastsq',
+                                x=x, data=data, flags=flags, wgts=wgts, output_prefix=output_prefix,
+                                cache=cache, ax=ax,
+                                skip_wgt=skip_wgt, max_contiguous_edge_flags=max_contiguous_edge_flags,
+                                verbose=verbose, overwrite=overwrite)
+    ###
+    # TODO: zeropad here will error given its default option if 2d filtering is being used.
```

Comment:
This should be an issue, not a TODO in the code. Otherwise we'll lose track of it.

Reply:
This is a stale TODO. Zeropadding is handled correctly now.